src/app/utils/BountyUtils.ts

```diff
@@ -262,6 +269,201 @@ const BountyUtils = {
         }
         return title;
     
+    },
+
+    async canonicalCard(bountyId: string, activity: string, bountyChannel?: TextChannel): Promise<Message> {
+        Log.debug(`Creating/updating canonical card`);
+
+        // Get the updated bounty
+        const db: Db = await MongoDbUtils.connect('bountyboard');
+        const bountyCollection = db.collection('bounties');
+        const bounty: BountyCollection = await bountyCollection.findOne({
+            _id: new mongo.ObjectId(bountyId)
+        });
+        const customerCollection = db.collection('customers');
+        const customer: CustomerCollection = await customerCollection.findOne({
+            customerId: bounty.customerId,
+        });
+
+        // Build the fields, reactions, and footer based on status
+        const fields = [
+            { name: 'Bounty Id', value: bounty._id.toString(), inline: false },
+            { name: 'Criteria', value: bounty.criteria.toString() },
+            { name: 'Reward', value: bounty.reward.amount + ' ' + bounty.reward.currency, inline: true },
+            { name: 'Status', value: bounty.status, inline: true },
+            { name: 'Deadline', value: BountyUtils.formatDisplayDate(bounty.dueAt), inline: true },
+            { name: 'Created by', value: bounty.createdBy.discordHandle.toString(), inline: true }
+        ];
+        if (bounty.gate) {
+            const role = await DiscordUtils.getRoleFromRoleId(bounty.gate[0], bounty.customerId);
+            fields.push({ name: 'For role', value: role.name, inline: false })
+        }
+        if (bounty.assign) {
+            const assignedUser = await DiscordUtils.getGuildMemberFromUserId(bounty.assign, bounty.customerId);
+            fields.push({ name: 'For user', value: assignedUser.user.tag, inline: false })
+        }
+    
+        let footer = {};
+        let reacts = [];
+        let color = undefined;
+
+        switch (bounty.status) {
+            case BountyStatus.draft:
+                footer = { text: '👍 - publish | ❌ - delete | Please reply within 60 minutes', };
+                reacts.push('👍');
+                reacts.push('❌');
+                break;
+            case BountyStatus.open:
+                if (bounty.requireApplication && (!bounty.assign)) {
+                    footer = { text: '🙋 - apply | ❌ - delete', };
+                    reacts.push('🙋');
+                } else {
+                    footer = { text: '🏴 - claim | ❌ - delete', };
+                    reacts.push('🏴');
+                }
+                reacts.push('❌');
+                break;
+            case BountyStatus.in_progress:
+                color = '#d39e00';
+                reacts.push('📮');
+                reacts.push('✅');
+                if (!bounty.paidStatus || bounty.paidStatus === PaidStatus.unpaid) {
```

Comment:
I think for this and those like it youmean to use `&&` instead of `||`

Reply:
Actually I think I mean || (or). If there is no status, OR it is unpaid, include the Paid reaction

Reply:
ah fair enough, yea

Then, a nit: In the case where we don't need to short circuit based on whether or not a value is null, we can remove the null check.

Reply:
Still learning my JS truthy and falsey rules :)